### PR TITLE
Branches/fix srv soa

### PR DESCRIPTION
**Pull request by cmikk**
Fix SRV encoding error, and missing SOA mname indexing.

### CODE REVIEW

File: dnstable_convert.c [hunk 1]
```diff
@@ -310,11 +310,19 @@ process_rdata_slice(Nmsg__Sie__DnsDedupe *dns, size_t i, ubuf *key, ubuf *val)
...
 	wdns_res res;
 
 	switch (dns->rrtype) {
+	case WDNS_TYPE_SOA:
+		res = wdns_len_uname(dns->rdata[i].data,
+				     dns->rdata[i].data + len,
+				     &len);
+		if (res != wdns_res_success)
+			return;
+		printf("SOA record found, len=%zd\n", len);
```
Comment (djw1149) [inline dnstable_convert.c]: is the printf leftover debugging code?

File: dnstable_convert.c [hunk 2]
```diff
@@ -310,11 +310,19 @@ process_rdata_slice(Nmsg__Sie__DnsDedupe *dns, size_t i, ubuf *key, ubuf *val)
...
 {
-	size_t offset;
+	size_t offset, len = dns->rdata[i].len;
 	uint8_t name[WDNS_MAXLEN_NAME];
 	wdns_res res;
 
 	switch (dns->rrtype) {
+	case WDNS_TYPE_SOA:
+		res = wdns_len_uname(dns->rdata[i].data,
+				     dns->rdata[i].data + len,
```
Comment (djw1149) [inline dnstable_convert.c]: this is using len before the check on its validity that is at line 344

File: dnstable_convert.c [hunk 3]
```diff
@@ -323,9 +331,11 @@ process_rdata_name_rev(Nmsg__Sie__DnsDedupe *dns, size_t i, ubuf *key, ubuf *val
 		break;
 	case WDNS_TYPE_MX:
 		offset = 2;
```
Comment (djw1149) [inline dnstable_convert.c]: please document where the magic numbers 0, 2, and 6 come from.

File: dnstable_convert.c [hunk 4]
```diff
@@ -323,9 +331,11 @@ process_rdata_name_rev(Nmsg__Sie__DnsDedupe *dns, size_t i, ubuf *key, ubuf *val
...
+		len -= offset;
 		break;
 	case WDNS_TYPE_SRV:
 		offset = 6;
+		len -= offset;
 		break;
 	default:
```
Comment (djw1149) [inline dnstable_convert.c]: is the default: case an error or are the other rrtypes that exist that we simply don't need to handle?

File: dnstable_convert.c (hunk 1 above)
Comment (cmikk) [reply dnstable_convert.c]: Yes, it is. Will fix.

File: dnstable_convert.c (hunk 4 above)
Comment (cmikk) [reply dnstable_convert.c]: The `default:` case here is that we do not create an index entry for the rdata.

Comment (djw1149) [reply dnstable_convert.c]: please add a comment for that.

File: dnstable_convert.c (hunk 2 above)
Comment (cmikk) [reply dnstable_convert.c]: I do not think a validity check (effectively, `if (len == 0) return;` is needed, nor would it add to the readability. We would merely be passing `wdns_len_uname()` a zero-length buffer, a case which it is expected to handle (and does).

Comment (djw1149) [reply dnstable_convert.c]: I hear you, but we have such a check later 
```C
if (dns->rdata[i].len == 0 || dns->rdata[i].len <= offset)
		return;
```

Comment (cmikk) [reply dnstable_convert.c]: Added check.

**Pull request merged**